realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -428,6 +418,11 @@ private void enableCollectionSnapshot() {
     }
 
     void disableCollectionSnapshot() {
+        if (isInTransaction()) {
+            // This should never happen.
+            throw new IllegalStateException( "Collection cannot be reattached if the Realm is in transaction." +
+                    " Please remember to commit or cancel transaction before finishing the current event loop.");
```

Comment:
Makes sense, I've never done that but was thinking about the possibility - now it's obviously not a possibility 😄